### PR TITLE
[IMP] base_(geolocalize): sync child geo location

### PR DESCRIPTION
**Pull request by mrsr-odoo**
Purpose
=======
Make it easier to retrieve the partner's coordinates based on the parent.

Specification
==============
- Sync partner coordinates with the parent when the type is in contact.
- It makes the coordinates field read-only and invisible computation buttons for the same.

Task-3336288

### CODE REVIEW

File: odoo/addons/base/models/res_partner.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 WARNING_HELP = 'Selecting the "Warning" option will notify user with the message, Selecting "Blocking Message" will throw an exception with the message and block the flow. The Message has to be written in the next field.'
 
 
-ADDRESS_FIELDS = ('street', 'street2', 'zip', 'city', 'state_id', 'country_id')
+ADDRESS_FIELDS = ('street', 'street2', 'zip', 'city', 'state_id', 'country_id', 'partner_latitude', 'partner_longitude')
```
Comment (awa-odoo) [inline odoo/addons/base/models/res_partner.py:29]: I think that means that "_formatting_address_fields" will now also return lat / lon.
See: https://github.com/odoo/odoo/blob/master/odoo/addons/base/models/res_partner.py#L573

We probably don't want that?

**Review by awa-odoo — COMMENTED** [review]
Hello @mrsr-odoo 

One question in the comments.

Thanks & cheers!

File: odoo/addons/base/models/res_partner.py (hunk 1 above)
Comment (mrsr-odoo) [reply odoo/addons/base/models/res_partner.py:29]: Hello @awa-odoo ,

If so, Can't we filter out for "_formatting_address_fields"
by adding this 
```
return [field for field in self._address_fields() if field not in ['partner_latitude', 'partner_longitude']]
```

Thanks

Comment (awa-odoo) [reply odoo/addons/base/models/res_partner.py:29]: Hello.

Yes, I think it would be better to avoid undesired behaviors.

**Comment by mrsr-odoo** [comment]
Hello @awa-odoo :slightly_smiling_face: ,

Updated the code 
Please have a look

Thanks !

**Comment by awa-odoo** [comment]
@robodoo r-

**Comment by awa-odoo** [comment]
Hello @mrsr-odoo 

Before the merge, can you complement your commit message with the following:

> Note
> ====
> We purposefully decided to not migrate existing contact to sync their longitude
> and latitude fields with their respective parent partners.
> 
> As these fields are not widely used and not critical, we consider it's not
> worth it to introduce potential performance bottlenecks in the migration, as
> res.partner is a big central table within the Odoo ecosystem.
> 
>  If people want to manually adapt their data, it can be done using
export/import.

That way we explain everything thoroughly.

Thanks!

**Comment by awa-odoo** [comment]
Plot twist: finally we don't really want to do it, not worth the risk, see task pad for details.

**Pull request closed without merging**